Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,792 @@
+        switch(webGLValue) {
+            case WebGLConstants.FLOAT:
+                return 'float';
+            case WebGLConstants.FLOAT_VEC2:
+                return 'vec2';
+            case WebGLConstants.FLOAT_VEC3:
+                return 'vec3';
+            case WebGLConstants.FLOAT_VEC4:
+                return 'vec4';
+            case WebGLConstants.FLOAT_MAT2:
+                return 'mat2';
+            case WebGLConstants.FLOAT_MAT3:
+                return 'mat3';
+            case WebGLConstants.FLOAT_MAT4:
+                return 'mat4';
+            case WebGLConstants.SAMPLER_2D:
+                return 'sampler2D';
+        }
+    }
+
+    function generateLightParameters(gltf) {
+        var result = {};
+
+        var lights;
+        if (defined(gltf.extensions) && defined(gltf.extensions.KHR_materials_common)) {
+            lights = gltf.extensions.KHR_materials_common.lights;
+        }
+
+        if (defined(lights)) {
+            // Figure out which node references the light
+            var nodes = gltf.nodes;
+            for (var nodeName in nodes) {
+                if (nodes.hasOwnProperty(nodeName)) {
+                    var node = nodes[nodeName];
+                    if (defined(node.extensions) && defined(node.extensions.KHR_materials_common)) {
+                        var nodeLightId = node.extensions.KHR_materials_common.light;
+                        if (defined(nodeLightId) && defined(lights[nodeLightId])) {
+                            lights[nodeLightId].node = nodeName;
+                        }
+                        delete node.extensions.KHR_materials_common;
+                    }
+                }
+            }
+
+            // Add light parameters to result
+            var lightCount = 0;
+            for(var lightName in lights) {
+                if (lights.hasOwnProperty(lightName)) {
+                    var light = lights[lightName];
+                    var lightBaseName = 'light' + lightCount.toString();
+                    light.baseName = lightBaseName;
+                    switch(light.type) {
+                        case "ambient":
+                        {
+                            var ambient = light.ambient;
+                            result[lightBaseName + 'Color'] =
+                            {
+                                type: WebGLConstants.FLOAT_VEC3,
+                                value: ambient.color
+                            };
+                        }
+                            break;
+                        case "directional":
+                        {
+                            var directional = light.directional;
+                            result[lightBaseName + 'Color'] =
+                            {
+                                type: WebGLConstants.FLOAT_VEC3,
+                                value: directional.color
+                            };
+                            if (defined(light.node)) {
```

Comment:
If a directional/point/spot light is not referenced by a node, then it doesn't need to be part of the shader since it is not actually in the scene, right?